### PR TITLE
Additional method test for some ML algos

### PR DESCRIPTION
**Pull request by WeichenXu123**
Add additional method test for some ML algos.

In this PR, I add `associationRules` in `FPGrowth` and `findSynonyms`. 
After the design is accepted, I will add other methods later.

Add an interface in `BenchmarkableAlgorithm`:
```
  def testAdditionalMethods(ctx: MLBenchContext, model: Transformer): Map[String, () => _]
```

### CODE REVIEW

**Comment by ludatabricks** [comment]
I have run the test manually. Everything is clean.

File: src/main/scala/com/databricks/spark/sql/perf/mllib/BenchmarkAlgorithm.scala [hunk 1]
```diff
@@ -44,6 +45,11 @@ trait BenchmarkAlgorithm extends Logging {
   def name: String = {
     this.getClass.getCanonicalName.replace("$", "")
   }
+
+  /**
+   * Test additional methods for some algorithms.
```
Comment (jkbradley) [inline src/main/scala/com/databricks/spark/sql/perf/mllib/BenchmarkAlgorithm.scala]: Can you please document what this should return?

File: src/main/scala/com/databricks/spark/sql/perf/mllib/BenchmarkAlgorithm.scala [hunk 2]
```diff
@@ -1,13 +1,14 @@
...
 
 import com.typesafe.scalalogging.slf4j.{LazyLogging => Logging}
-
 import org.apache.spark.ml.attribute.{NominalAttribute, NumericAttribute}
 import org.apache.spark.ml.{Estimator, PipelineStage, Transformer}
 import org.apache.spark.ml.evaluation.Evaluator
 import org.apache.spark.sql._
 import org.apache.spark.sql.functions._
 
+import com.databricks.spark.sql.perf.MLResult
```
Comment (jkbradley) [inline src/main/scala/com/databricks/spark/sql/perf/mllib/BenchmarkAlgorithm.scala]: unneeded?

File: src/main/scala/com/databricks/spark/sql/perf/mllib/feature/Word2Vec.scala [hunk 3]
```diff
@@ -31,4 +36,17 @@ object Word2Vec extends BenchmarkAlgorithm with TestFromTraining {
     new ml.feature.Word2Vec().setInputCol("text")
   }
 
+  override def testAdditionalMethods(ctx: MLBenchContext, model: Transformer)
```
Comment (jkbradley) [inline src/main/scala/com/databricks/spark/sql/perf/mllib/feature/Word2Vec.scala]: style: incorrect format for multi-line method header  (here and elsewhere)

File: src/main/scala/com/databricks/spark/sql/perf/mllib/fpm/FPGrowth.scala [hunk 4]
```diff
@@ -28,4 +29,13 @@ object FPGrowth extends BenchmarkAlgorithm with TestFromTraining {
     new ml.fpm.FPGrowth()
       .setItemsCol("items")
   }
+
+  override def testAdditionalMethods(ctx: MLBenchContext, model: Transformer)
+      : Map[String, () => _] = {
+
+    val fpModel = model.asInstanceOf[FPGrowthModel]
+    Map("associationRules" -> (() => {
+      fpModel.associationRules
```
Comment (jkbradley) [inline src/main/scala/com/databricks/spark/sql/perf/mllib/fpm/FPGrowth.scala]: I think this is lazy, so please add a count() here to force materialization

**Comment by jkbradley** [comment]
LGTM
The tests worked when run locally.
Thanks @WeichenXu123 !
Merging with master

**Pull request merged**